controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -833,8 +839,12 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 	// Create the hostFirmwareSettings resource with same host name/namespace if it doesn't exist
 	if info.host.Name != "" {
 		if err = r.createHostFirmwareSettings(info); err != nil {
-			info.log.Info("failed creating hostfirmwaresettings")
-			return actionError{errors.Wrap(err, "failed creating hostFirmwareSettings")}
+			if r.ignoreErrorWhenCleaningModeDisabled(err, info) {
```

Comment:
I don't see any point trying to create the HostFirmwareSettings when the BareMetalHost already has a DeletionTimestamp. Let's not do it in all circumstances where deletion has started, instead of always doing it and ignoring the error in a narrow set of circumstances.